### PR TITLE
EPAM-5: Aws S3

### PR DESCRIPTION
**Pull request by musiichuk**
Link to commit for fe part (https://github.com/musiichuk/nodejs-aws-fe/commit/facf736c3a77df42fc0f794fd7b4a747f8d13757)

Link to Front End App For Checking uploading: https://d15tir2c449mlg.cloudfront.net/admin/products

image

### CODE REVIEW

File: import-service/src/handlers/importFileParser.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const AWS = require('aws-sdk');
+const csvParser = require('csv-parser')
+
+const BUCKET = 'products-bucket-aws-epam';
+
+// const moveToParsedFolder = (event) => {
```
Comment (MaxPlav) [inline import-service/src/handlers/importFileParser.js:6]: why is it commented? any troubles?
btw, the `moveToParsedFolder` function should be `async`

File: import-service/src/handlers/importFileParser.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const s3Stream = s3.getObject(params).createReadStream();
+    const csvParserStream = csvParser();
+
+    s3Stream
+        .pipe(csvParserStream)
+        .on('data', (data) => console.log('DATA: ', data))
+        .on('error', (error) => console.log('ERROR importFileParser: ', error))
+        .on('end', () => console.log('END'))
+
+    // moveToParsedFolder(event);
```
Comment (MaxPlav) [inline import-service/src/handlers/importFileParser.js:41]: calling of `moveToParsedFolder` should be at the end of the stream - on 'end' handler

File: import-service/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "import-service",
+  "version": "1.0.0",
+  "description": "",
+  "main": "handler.js",
+  "dependencies": {
+    "aws-sdk": "^2.899.0",
+    "csv-parser": "^3.0.0",
+    "jest": "^26.6.3",
+    "serverless-offline": "^7.0.0"
```
Comment (ikushlianski) [inline import-service/package.json:10]: should be in devDependencies, as well as `jest`

File: import-service/src/handlers/importFileParser.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const AWS = require('aws-sdk');
+const csvParser = require('csv-parser')
+
+const BUCKET = 'products-bucket-aws-epam';
```
Comment (ikushlianski) [inline import-service/src/handlers/importFileParser.js:4]: better be moved to env variables in a real project

File: import-service/src/handlers/importProductsFile.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const csvFilename = path.resolve(__dirname, 'test.csv');
+
+    it('should return a correct status code and body must contain a domain to bucket', async () => {
+        const event = {
+            queryStringParameters: {
+                name: 'test.csv'
+            }
+        }
+
+        awsMock.mock("S3", "putObject", Buffer.from(require("fs").readFileSync(csvFilename)));
```
Comment (ikushlianski) [inline import-service/src/handlers/importProductsFile.test.js:18]: unit tests should not access the filesystem, so I suggest mocking this one too

**Pull request merged**